kirovy/models/cnc_game.py

```diff
@@ -3,10 +3,67 @@
 from django.conf import settings
 from django.db import models
 
-from kirovy import exceptions
+from kirovy import exceptions, typing as t
 
 from kirovy.models.cnc_base_model import CncNetBaseModel
 
+__all__ = ["CncFileExtension", "CncGame"]
+
+
+def is_valid_extension(extension: str) -> None:
+    """Validate file extension string.
+
+    :raise exceptions.InvalidFileExtension:
+        Raised for invalid file extension strings.
+    """
+    if not extension.isalnum():
+        raise exceptions.InvalidFileExtension(
+            f'"{extension}" is not a valid file extension. Must be alpha only.'
+        )
+
+
+class CncFileExtension(CncNetBaseModel):
```

Comment:
It makes it easier to add support for generals down the line

Reply:
How exactly do you mean? I just thought maybe storing allowed extensions as a string would be much simpler to work with.

Also how about not sticking to the extension but to the actual *type* instead? like YRM, MPR, MAP are actually duplicates just for different games. Not sure though how that would look.

Reply:
It allows for better querying and makes it easier to support stuff without code changes. E.g. if we wanted to allow zip files at some point. 

I don't really like have string constants for things that should be changeable. 

E.g. to add support for generals under this paradigm, you'd just need to log into Django admin and add .big and .str to the table.

Reply:
No no, I actually was the one who said against a const lol, I meant that maybe we should just store some sort of an array in a column? Not sure if there are elegant solutions but still. Also what about the "stick to actual inner type instead of an extension"? Not sure of use but still, just voicing my thoughts.

Reply:
Oh I'll actually check the content of the file instead of just checking the extension. I get what you're saying though. I guess I just don't like having arrays in a column of a database.